server/space-router.go

```diff
@@ -123,18 +129,24 @@ func (router *SpaceRouter) getAvailability(w http.ResponseWriter, r *http.Reques
 		m.Rotation = e.Rotation
 		m.Available = e.Available
 		m.Bookings = []*GetSpaceAvailabilityBookingsResponse{}
-		if showNames {
-			for _, booking := range e.Bookings {
-				enter, _ := attachTimezoneInformation(booking.Enter, location)
-				leave, _ := attachTimezoneInformation(booking.Leave, location)
-				entry := &GetSpaceAvailabilityBookingsResponse{
-					UserID:    booking.UserID,
-					UserEmail: booking.UserEmail,
-					Enter:     enter,
-					Leave:     leave,
-				}
-				m.Bookings = append(m.Bookings, entry)
+		for _, booking := range e.Bookings {
+			var showName bool = showNames
+			enter, _ := attachTimezoneInformation(booking.Enter, location)
+			leave, _ := attachTimezoneInformation(booking.Leave, location)
+			outUserId := ""
+			outUserEmail := ""
+			if showName || user.Email == booking.UserEmail {
+				outUserId = booking.UserID
+				outUserEmail = booking.UserEmail
+			}
+			entry := &GetSpaceAvailabilityBookingsResponse{
+				BookingID: booking.ID,
```

Comment:
After renaming ID to BookingID in space-repository.go, this should become:
`BookingID: booking.BookingID,`